src/SFA.DAS.LevyTransferMatching/Application/Queries/Pledges/GetApplications/GetApplicationsQueryHandler.cs

```diff
@@ -37,12 +41,8 @@ public async Task<GetApplicationsQueryResult> Handle(GetApplicationsQuery reques
                 var roles = roleReferenceData.Where(x => pledgeResponse.JobRoles.Contains(x.Id));
 
                 var applicationResult = (GetApplicationsQueryResult.Application)application;
-
-                applicationResult.IsLocationMatch = !pledgeResponse.Locations.Any() || application.Locations.Any();
-                applicationResult.IsSectorMatch = !pledgeResponse.Sectors.Any() || application.Sectors.Any(x => pledgeResponse.Sectors.Contains(x));
-                applicationResult.IsJobRoleMatch = !pledgeResponse.JobRoles.Any() || roles.Any(r => r.Description == application.StandardRoute);
-                applicationResult.IsLevelMatch = !pledgeResponse.Levels.Any() || pledgeResponse.Levels.Select(x => char.GetNumericValue(x.Last())).Contains(application.StandardLevel);
-
+                
+                AddPledgeDetailsToApplication(applicationResult, pledgeResponse, application, roles);
```

Comment:
As much as I like the implicit operators, I think this stitching together of these bits is a bit messy (i.e. using the implicit operator to do "most of" the mapping and then doing a bit more elsewhere), though I appreciate the "match" fields were being tacked on like that previously.

Could we just remove the implicit operator here and add a mapping method which takes all the required params so the application result gets instantiated and all properties set in one go?

Reply:
added a factory method that takes the parameters and builds the object